app/models/concerns/hostext/search.rb

```diff
@@ -171,6 +171,20 @@ def search_by_config_group(key, operator, value)
         {:conditions => opts, :include => :hostgroup}
       end
 
+      def cast_facts(key, operator, value)
+        begin
+          is_int = !!Integer(value)
```

Comment:
```ruby
n = 10_000
Benchmark.bm do |x|
  x.report { n.times do; '123' =~ /\A[-+]?\d+\z/; end }
  x.report { n.times do; '123'.to_i.to_s == '123'; end }
  x.report { n.times do; !!Integer('123'); end }
end
```
```
       user     system      total        real
   0.010000   0.000000   0.010000 (  0.005614)
   0.010000   0.000000   0.010000 (  0.007447)
   0.000000   0.000000   0.000000 (  0.004172)
```

you're right, it's slightly faster but in this case we don't do the operation for 10_000 records but for 1 which IMHO does not justify using exceptions for driving program flow